pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -498,7 +585,7 @@ func GetPipelineConditionStatus(pr *v1beta1.PipelineRun, state PipelineRunState,
 
 	// Hasn't timed out; not all tasks have finished....
 	// Must keep running then....
-	if failedTasks > 0 || cancelledTasks > 0 {
+	if cancelledTasks > 0 || (failedTasks > 0 && state.isFinalTasksDone(dfinally)) {
```

Comment:
I don't understand this. If all final tasks are done, the pipeline is not "stopping", that state only happens when the controller is waiting for taskruns to complete executing *before* finally tasks are started.

Reply:
question: what state is the pipeline supposed to be in when final tasks are running, is it running or is it stopping?

i would assume running, unless something failed or was cancelled

so i would think this block doesn't need to change?

Reply:
this additional check is making sure that pipeline does not exit when final tasks are not done executing.

without this additional check `state.isFinalTasksDone(dfinally)`, `GetPipelineConditionStatus` changes the pipeline status to `stopping` irrespective of where the final tasks are in their execution cycle (scheduled, started, running, failed, succeeded, retrying etc)

The pipeline is in `running` state when final tasks are running. If we do not change this block, pipeline status changes to `stopping` as soon as it encounters a DAG failure. 

`GetPipelineConditionStatus` function is the next immediate candidate for refactoring

Reply:
Thanks for the explanation.